src/Microsoft.Diagnostics.Monitoring.RestServer/Controllers/DiagController.cs

```diff
@@ -75,26 +73,18 @@ public Task<ActionResult> GetGcDump(int? pid)
             });
         }
 
-        [HttpGet("cpuprofile/{pid?}")]
-        public Task<ActionResult> CpuProfile(int? pid, [FromQuery][Range(-1, int.MaxValue)]int durationSeconds = 30)
-        {
-            TimeSpan duration = ConvertSecondsToTimeSpan(durationSeconds);
-            return InvokeService(async () =>
-            {
-                int pidValue = _diagnosticServices.ResolveProcess(pid);
-                IStreamWithCleanup result = await _diagnosticServices.StartCpuTrace(pidValue, duration, this.HttpContext.RequestAborted);
-                return new StreamWithCleanupResult(result, "application/octet-stream", FormattableString.Invariant($"{Guid.NewGuid()}.nettrace"));
-            });
-        }
-
         [HttpGet("trace/{pid?}")]
-        public Task<ActionResult> Trace(int? pid, [FromQuery][Range(-1, int.MaxValue)]int durationSeconds = 30)
+        public Task<ActionResult> Trace(
+            int? pid,
+            [FromQuery]TraceProfile profile = TraceProfile.All,
```

Comment:
Should the default be all?

There's a sizeable observability impact for logging right now (especially since the EventSourceLogger hasn't yet been optimized) which can skew whatever you're trying to observe via CPU profiling

Reply:
There are three options:
- There's no default; invoking the endpoint without the profile parameter is an error.
- Some subset of profiles that are default that the user will have to look up some type of documentation for to know what the default consists.
- All profiles are enabled. Again, the user will probably have to look up some type of documentation to know what they are, but at least they'll know that it's all of the profiles rather than some subset.

Previously, this endpoint would return the http requests and logging, so it had the same exact problem that you are already bringing up. The default now includes cpu and metrics in addition to the what it was previously providing.

Reply:
I think my main concern is that turning on both logging and CPU profiling doesn't lend it self well to any perf investigation right now. Our defaults should set you up for success.

My suggestion is omit logs from the default.